From: SeongJae Park <sj@kernel.org>
Subject: mm/damon/core: initialize sidx in damos_trace_esz()
Date: Sat, 5 Jul 2025 10:20:03 -0700

Commit ae4dc5a31d0f ("mm/damon: add trace event for effective size
quota") introduced damos_trac_esz(), which uses sidx variable without
initialization.  Initialize before using it.

Link: https://lkml.kernel.org/r/20250705172003.52324-1-sj@kernel.org
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202507051517.bs3jd8RW-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
Cc: "Masami Hiramatsu (Google)" <mhiramat@kernel.org>
Cc: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Cc: Steven Rostedt <rostedt@goodmis.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/damon/core.c~mm-damon-add-trace-event-for-effective-size-quota-fix
+++ a/mm/damon/core.c
@@ -2014,7 +2014,7 @@ static void damos_set_effective_quota(st
 static void damos_trace_esz(struct damon_ctx *c, struct damos *s,
 		struct damos_quota *quota)
 {
-	unsigned int cidx = 0, sidx;
+	unsigned int cidx = 0, sidx = 0;
 	struct damos *siter;
 
 	damon_for_each_scheme(siter, c) {
_
